corehq/apps/app_manager/suite_xml.py

```diff
@@ -1998,6 +1998,7 @@ def avoid_duplicate_ids(action_, this_datum_, parent_datum_, datum_ids_, changed
                 if root_module_form.requires_case():
                     root_datums.extend(self.get_datum_meta_module(root_module))
                 root_datums.extend(self.get_new_case_id_datums_meta(root_module_form))
+                root_datums.extend(self.get_extra_case_id_datums(root_module_form))
```

Comment:
Seems like we could extract these 4 lines into a function since they are repeated exactly here: https://github.com/dimagi/commcare-hq/blob/autostaging/corehq/apps/app_manager/suite_xml.py#L1705-1705